src/timezone_tools/_converter.py

```diff
@@ -252,9 +252,11 @@ def start_of_month(
                 timezone to localize from. Use `make_aware` to make a naive
                 datetime timezone-aware.
         """
-        return self.midnight(self.first_day_of_month(datetime))
+        return self.midnight(self.first_day_of_month(when))
 
-    def end_of_month(self, datetime: datetime_.datetime) -> datetime_.datetime:
+    def end_of_month(
+        self, when: datetime_.datetime | datetime_.date
```

Comment:
Changing the arguments of a public function is a breaking change. Could we avoid doing that?

Reply:
Ofcourse. I was trying to follow the pattern of the other methods that use `when` when both `date` and `datetime` are allowed, but we can ofcourse use `datetime`, although I find that a bit confusing. I'll extend the docstring to make it clearer maybe.

Reply:
Yeah it is a bit strange. This library is fairly knew and we're still on v0, so it's probably okay to change these if we add it to the change log (and, of course, so do in its own commit)